### PR TITLE
Redefine range() in Python 2

### LINKED ISSUE #130: xrange() was removed in Python 3
flake8 testing of https://github.com/opencv/open_model_zoo on Python 3.7.1

$ __flake8 . --count --select=E9,F63,F72,F82 --show-source --statistics__
```
./demos/smart_classroom_demo/action_event_metrics.py:94:33: F821 undefined name 'xrange'
            for bbox_attr_id in xrange(len(bbox)):
                                ^
./demos/smart_classroom_demo/action_event_metrics.py:172:18: F821 undefined name 'xrange'
        for i in xrange(len(sorted_predicted_bboxes)):
                 ^
./demos/smart_classroom_demo/action_event_metrics.py:178:26: F821 undefined name 'xrange'
            for gt_id in xrange(len(gt_bboxes)):
                         ^
./demos/smart_classroom_demo/action_event_metrics.py:277:22: F821 undefined name 'xrange'
            for i in xrange(1, len(input_events)):
                     ^
./demos/smart_classroom_demo/action_event_metrics.py:327:29: F821 undefined name 'xrange'
            for event_id in xrange(1, len(input_events)):
                            ^
./demos/smart_classroom_demo/action_event_metrics.py:385:20: F821 undefined name 'xrange'
    for pred_id in xrange(len(pred_events)):
                   ^
./demos/smart_classroom_demo/action_event_metrics.py:388:22: F821 undefined name 'xrange'
        for gt_id in xrange(len(gt_events)):
                     ^
7     F821 undefined name 'xrange'
```

### PR DESCRIPTION
**Pull request by cclauss**
__xrange()__ was removed in Python 3 in favor of a re-worked version of __range()__.  This PR assures equivalent functionality in both Python 2 and Python 3.  Fixes #130

### CODE REVIEW

File: demos/smart_classroom_demo/action_event_metrics.py [hunk 1]
```diff
@@ -23,6 +23,11 @@
 from lxml import etree
 from tqdm import tqdm
 
+try:
+    xrange          # Python 2
+except NameError:
+    xrange = range  # Python 3
+
```
Comment (evgeny-izutov) [inline demos/smart_classroom_demo/action_event_metrics.py]: @cclauss, I think the more elegant way to fix the compatibility issue is to add `from builtins import range` import row and replace all `xrange` onto `range`.

Comment (cclauss) [reply demos/smart_classroom_demo/action_event_metrics.py]: Changed but without a requirements.txt or setup.py it was difficult to determine if _future_ or six were installed dependencies so I did it the old fashioned way.

File: demos/smart_classroom_demo/action_event_metrics.py [hunk 2]
```diff
@@ -174,13 +170,13 @@ def match_detections(predicted_data, gt_data, min_iou):
 
         matches = []
         visited_gt = np.zeros(len(gt_bboxes), dtype=np.bool)
-        for i in xrange(len(sorted_predicted_bboxes)):
+        for i in range(len(sorted_predicted_bboxes)):
```
Comment (evgeny-izutov) [inline demos/smart_classroom_demo/action_event_metrics.py]: According usage of `range` function in the code it looks like we can eliminate the `range` function at all: replace `for i in range(len())` construction onto `for i, _ in enumerate()` and so on.

Comment (cclauss) [reply demos/smart_classroom_demo/action_event_metrics.py]: I am a huge fan of __enumerate()__ but it’s use is too counter intuitive if we are not going to use each element.  My preference would be to stick with your first solution instead.

Comment (evgeny-izutov) [reply demos/smart_classroom_demo/action_event_metrics.py]: Agree.

**Comment by yssemaev** [comment]
Hi, per branching strategy, please merge to "develop" branch (not "master")

**Comment by kchechil** [comment]
Jenkins please retry a build

**Pull request merged**